webapp/channels/src/utils/utils.tsx

```diff
@@ -212,6 +212,31 @@ const removeQuerystringOrHash = (extin: string): string => {
 };
 
 export const getFileType = (extin: string): typeof FileTypes[keyof typeof FileTypes] => {
+    // Handle null or undefined input
+    if (!extin) {
+        return FileTypes.OTHER;
+    }
+
+    // Special handling for image proxy URLs
+    // Check for various forms of image proxy URLs
+    if (extin.includes('/api/v4/image') &&
+        (extin.includes('?url=') || extin.includes('&url='))) {
+        return FileTypes.IMAGE;
+    }
+
+    // Check for image file extensions in the URL path
+    const urlPath = extin.split('?')[0]; // Remove query parameters
+    const pathParts = urlPath.split('/');
+    const lastPathPart = pathParts[pathParts.length - 1];
+
+    if (lastPathPart && lastPathPart.includes('.')) {
+        const urlExtension = lastPathPart.split('.').pop()?.toLowerCase();
+        if (urlExtension && Constants.IMAGE_TYPES.indexOf(urlExtension) > -1) {
+            return FileTypes.IMAGE;
+        }
+    }
```

Comment:
Is `extin` guaranteed to be URL? I realize `split` will always return a non-empty array in these cases, though it's not super obvious that the code is "safe" in that regards.

Would it make sense to try to parse it as a `URL` and use methods to examine the extension? (And if we can't parse it, then continue with the rest of the code.)

Reply:
ok, in my next commit will use url build in parse function and wrap this in a try-catch